client/src/modules/profiles/pages/userProfile.tsx

```diff
@@ -59,6 +85,24 @@ export const UserProfilePage = () => {
           <Heading as="h2" size={'lg'}>
             Welcome {user.name}
           </Heading>
+          <FormControl>
+            <Flex>
+              <FormLabel htmlFor="subscription">
+                Automatic subscription
+              </FormLabel>
+              <Switch
+                id="subscription"
+                isChecked={autoSubscribe}
+                onChange={onSwitch}
+                isDisabled={isLoading}
+              />
+              {isLoading && <Spinner size="sm" />}
```

Comment:
I think the spinner is redundant with `isDisabled={isLoading}` because the switch color is changed subtly when it's disabled, and the cursor changes as well

```suggestion
```

Reply:
I found the difference between disabled and unchecked to not be that distinct. It can be also disabled just for fractions of second, at which point if somebody isn't specifically looking for it, it might not be clear that switch was disabled. Appearing spinner is more likely to catch eye.

I'd consider disabling it only as a way to prevent from multiple clicks before setting is changed, and the spinner (or something else) as an indicator that something is happening/happened.

What do you think?

Reply:
Some people will find getting email, annoying, and it will be more annoying to think you are disabled it, but it actually didn't because the request wasn't sent to the server, and you had no idea of known that. 🤔 👍 

I am convinced